core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -554,7 +726,11 @@ private[spark] class MapOutputTrackerMaster(
   def containsShuffle(shuffleId: Int): Boolean = shuffleStatuses.contains(shuffleId)
 
   def getNumAvailableOutputs(shuffleId: Int): Int = {
-    shuffleStatuses.get(shuffleId).map(_.numAvailableOutputs).getOrElse(0)
+    shuffleStatuses.get(shuffleId).map(_.numAvailableMapOutputs).getOrElse(0)
+  }
+
+  def getNumAvailableMergeResults(shuffleId: Int): Int = {
```

Comment:
Is this necessary? (except the tests)

Reply:
This is only used in test, will clarify.